File: backup/labs/compliance-portal/app/backend/app/cache/cache_manager.py
import time

class CacheManager:
    def __init__(self):
        self.cache = {}
        self.ttl = 60 # 1 minute in seconds

    def set(self, key: str, value: any, custom_ttl: int = None):
        expires_at = time.time() + (custom_ttl or self.ttl)
        self.cache[key] = {"value": value, "expires_at": expires_at}

    def get(self, key: str) -> any:
        entry = self.cache.get(key)
        if not entry:
            return None
        if time.time() > entry["expires_at"]:
            del self.cache[key]
            return None
        return entry["value"]

    def invalidate(self, key: str):
        if key in self.cache:
            del self.cache[key]

    def invalidate_all(self):
        self.cache.clear()

cache_manager = CacheManager()

File: backup/labs/compliance-portal/app/server/cache/cacheManager.js
class CacheManager {
  constructor() {
    this.cache = new Map();
    this.ttl = 60 * 1000; // 1 minute TTL by default
  }

  set(key, value, customTtl) {
    const expiresAt = Date.now() + (customTtl || this.ttl);
    this.cache.set(key, { value, expiresAt });
  }

  get(key) {
    const entry = this.cache.get(key);
    if (!entry) return null;
    if (Date.now() > entry.expiresAt) {
      this.cache.delete(key);
      return null;
    }
    return entry.value;
  }

  invalidate(key) {
    if (key) {
      this.cache.delete(key);
    } else {
      this.cache.clear();
    }
  }

  invalidateAll() {
    this.cache.clear();
  }
}

export const cacheManager = new CacheManager();
